app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = current_user.comments.new(post_params)
+    @comment.post_id = params[:post_id]
+    @comment.save
+    redirect_to user_post_path(@comment.author_id, @comment.post_id)
```

Comment:
- [ ] in this method you should check if the comment was saved like below:
```
    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
```

Reply:
Thanks for the review, but I don't understand the else block code  @article = Post.find(params[:post_id])
      redirect_to post_path(@post),.....What does this do and where is the @post in post_path(@post) coming from?

Reply:
@mikethreels And what about the @article.

Reply:
Hi @bobb-Rob, sorry for the delay, I see you solved the issue in the meantime as your project is approved. Congratulations!!